src/goose/brokers/redis/batch.clj

```diff
@@ -0,0 +1,191 @@
+(ns ^:no-doc goose.brokers.redis.batch
+  (:require
+    [goose.batch :as batch]
+    [goose.brokers.redis.commands :as redis-cmds]
+    [goose.defaults :as d]
+    [goose.job :as job]
+    [goose.metrics :as m]
+    [goose.retry]
+    [goose.utils :as u]
+
+    [taoensso.carmine :as car]))
+
+(defn batch-keys [id]
+  {:batch-hash   (d/prefix-batch id)
+   :enqueued-set (str (d/prefix-batch id) "/enqueued")
+   :retrying-set (str (d/prefix-batch id) "/retrying")
+   :success-set  (str (d/prefix-batch id) "/success")
+   :dead-set     (str (d/prefix-batch id) "/dead")})
+
+(defn enqueue
+  [redis-conn {:keys [id jobs] :as batch}]
+  (let [{:keys [batch-hash enqueued-set]} (batch-keys id)
+        pruned-batch-state (dissoc batch :jobs)
+        job-ids (map :id jobs)]
+    (redis-cmds/atomic
+      redis-conn
+      (car/multi)
+      (car/hmset* batch-hash pruned-batch-state)
+      (apply car/sadd enqueued-set job-ids)
+      (doseq [job jobs]
+        (car/lpush (:ready-queue job) job)))))
+
+;;; When fetching a hash from Redis, primitive data types
+;;; like integer, boolean and keyword get converted to string.
+;;; We restore data types to avoid any type-casting errors.
+(defn- restore-data-types
+  [{:keys [linger-sec total status created-at] :as batch}]
+  (assoc batch
+    :linger-sec (Integer/valueOf linger-sec)
+    :total (Integer/valueOf total)
+    :status (keyword status)
+    :created-at (Long/valueOf created-at)))
+
+(defn get-batch
+  [redis-conn id]
+  (let [{:keys [batch-hash enqueued-set retrying-job-set success-set dead-set]} (batch-keys id)
+        [batch enqueued retrying success dead] (redis-cmds/wcar*
+                                                 redis-conn
+                                                 (car/parse-map (car/hgetall batch-hash) :keywordize)
```

Comment:
@siripr4 I'm using `car/parse-map` instead of `flat-sequence->map`, as it performs same function.